chain/blocksync/blocksync_client.go

```diff
@@ -264,6 +269,10 @@ func (bs *BlockSync) sendRequestToPeer(ctx context.Context, p peer.ID, req *Bloc
 }
 
 func (bs *BlockSync) processBlocksResponse(req *BlockSyncRequest, res *BlockSyncResponse) ([]*types.TipSet, error) {
+	if len(res.Chain) == 0 {
```

Comment:
this looks good. If it doesn't end up doing it, next step would be to set a target based on best peer performance (could be an attack vector but i think it's ok to assume at least one honest peer and update to not number of blocks sent but number of blocks verified in best chain if needed (quite a bit more complex))